source/server/hot_restarting_child.cc

```diff
@@ -82,6 +83,104 @@ void HotRestartingChild::terminateParent() {
   wrapped_request.mutable_request()->mutable_terminate();
   sendHotRestartMessage(parent_address_, wrapped_request);
   parent_terminated_ = true;
+  // Once setting parent_terminated_ == true, we can send no more hot restart RPCs, and therefore
+  // receive no more responses, including stats. So, now safe to forget our stat transferral state.
+  parent_counter_values_.clear();
+  parent_gauge_values_.clear();
+}
+
+void HotRestartingChild::mergeParentStats(Stats::Store& stats_store,
+                                          const HotRestartMessage::Reply::Stats& stats_proto) {
+  const std::unordered_map<std::string, CombineLogic> combine_logic_exceptions{
```

Comment:
Can you CONSTRUCT_ON_FIRST_USE this / make it static?

Reply:
I could, but then it would stay in memory forever. In terms of overall efficiency I think it might be better to just build it every time it's briefly+rarely used.

Reply:
It's going to happen every merge interval which defaults to 5s, seems worth it to just make it static, but don't feel that strongly about it.

Reply:
It will only happen while the hot restart parent is still alive. Once the parent is gone, the restarter_.getParentStats() will start returning nullptr, and restarter_.mergeParentStats() (the function containing this map) will no longer be called.

Reply:
OK either way.